### DISCUSSION TITLE
Feature Request: Fluid Typography

### DISCUSSION THREAD

**Discussion by developedbygeo** [tailwindlabs/tailwindcss#8087] [Ideas]
**Why?** 

TailwindCSS focuses on the mobile-first approach, using relative units for font sizes (among others). I think that having default options for fluid typography in place would be a great feature for responsiveness out of the box. Moreover, it would eliminate the need for additional packages or custom implementations, and compatibility-wise, it could be provided as an opt-in feature. 

**Compatibility**

Currently, caniuse.com indicates that  95% of the browsers support [viewport units](https://caniuse.com/viewport-units), with the respective percentage being 90% for the[ `clamp()` function](https://caniuse.com/?search=clamp). 

I reckon that this would lead to less duplication and leaner classes while maintaining Tailwind's customizable nature.

**Comment by taviks** [comment]
Until this is supported (if it ever is), can anyone point me to some examples of how to achieve a fluid typography and spacing system with Tailwind I'd love for it to "just work" with existing Tailwind classes... seems possible but some example code would be super helpful.

**Reply by wongjn** [reply]
You should be able to add fluid values your Tailwind configuration as you'd like:

```js
module.exports = {
  theme: {
    extend: {
      spacing: {
        1: 'clamp(…)',
        2: 'clamp(…)',
        3: 'clamp(…)',
        // etc…
      },
      fontSize: {
        xs: 'clamp(…)',
        sm: ['clamp(…)', { lineHeight: '1.125' }],
        base: ['clamp(…)', { lineHeight: '1.125' }],
        // etc…
      },
    },
  },
};

**Reply by philwolstenholme** [reply]
You can grab some code from https://cube-css-with-tailwind.netlify.app or https://christianpenrod.com/blog/tailwindcss-responsive-design-without-breakpoints.

**Reply by philwolstenholme** [reply]
Or use the calculators on https://utopia.fyi or https://chrisburnell.com/clamp-calculator

**Reply by barvian** [reply]
> can anyone point me to some examples of how to achieve a fluid typography and spacing system with Tailwind I'd love for it to "just work" with existing Tailwind classes

I felt the same way! I just released [a fluid plugin](https://github.com/barvian/fluid-tailwind) that works with every built-in utility and uses existing scales, i.e.:

https://github.com/tailwindlabs/tailwindcss/assets/868352/096c2220-1c77-4d43-9b6f-5d9d17343752

Curious to hear what others think 🙂

**Comment by adrienpoly** [comment]
I have implemented responsive font in several projects using [RFS](https://github.com/twbs/rfs)

There is an existing plugin by the creator of RFS and a discussion here https://github.com/tailwindlabs/tailwindcss/discussions/12238

I usually add it manually to my Tailwind config like that : 
```js
 fontSize: {
        xs: ['0.75rem', { lineHeight: '1rem' }],
        sm: ['0.875rem', { lineHeight: 'rfs(1.25rem)' }],
        base: ['rfs(1rem)', { lineHeight: 'rfs(1.5rem)' }],
        lg: ['rfs(1.125rem)', { lineHeight: 'rfs(1.75rem)' }],
        xl: ['rfs(1.25rem)', { lineHeight: 'rfs(1.75rem)' }],
        '2xl': ['rfs(1.5rem)', { lineHeight: 'rfs(2rem)' }],
        '3xl': ['rfs(1.875rem)', { lineHeight: 'rfs(2.25rem)' }],
        '4xl': ['rfs(2.25rem)', { lineHeight: 'rfs(2.5rem)' }],
        '5xl': ['rfs(3rem)', { lineHeight: '1' }],
        '6xl': ['rfs(3.75rem)', { lineHeight: '1' }],
        '7xl': ['rfs(4.5rem)', { lineHeight: '1' }],
        '8xl': ['rfs(6rem)', { lineHeight: '1' }],
        '9xl': ['rfs(8rem)', { lineHeight: '1' }]
      },
```


example PR : https://github.com/parisrb/paris-rb.org/pull/235
